components/butterfly/src/rumor/dat_file.rs

```diff
@@ -358,42 +341,62 @@ impl DatFile {
 /// file containing rumors exchanged by the butterfly server.
 #[derive(Debug, Default, PartialEq)]
 pub struct Header {
-    pub member_len:         u64,
-    pub service_len:        u64,
-    pub service_config_len: u64,
-    pub service_file_len:   u64,
-    pub election_len:       u64,
-    pub update_len:         u64,
-    pub departure_len:      u64,
+    offsets: HashMap<String, u64>,
 }
 
 impl Header {
     pub fn from_file<R>(reader: &mut R, version: u8) -> io::Result<(u64, Self)>
         where R: Read
     {
         let mut bytes = match version {
-            1 => vec![0; mem::size_of::<Self>()],
-            _ => vec![0; mem::size_of::<Self>() + 8],
+            1 => vec![0; HEADER_VERSION_1_SIZE],
+            _ => vec![0; HEADER_VERSION_2_SIZE],
```

Comment:
This should probably be
```suggestion
            2 => vec![0; HEADER_VERSION_2_SIZE],
            _ => unimplemented!(),
```
Or perhaps something a bit more user-friendly.

Reply:
Ah, yes, good thinking.